### PR TITLE
New package: RaptorTech.RaptorHardwareService version 2.3.6.0

### PR DESCRIPTION
**Pull request by chrisg32**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/153243)

### CODE REVIEW

**Comment by chrisg32** [comment]
@microsoft-github-policy-service agree

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> [0C14:057C][2024-05-12T07:30:02]e000: Error 0x800700d8: Process returned error: 0xd8
> [0C14:057C][2024-05-12T07:30:02]e000: Error 0x800700d8: Failed to execute EXE package.
> [0F48:1064][2024-05-12T07:30:02]e000: Error 0x800700d8: Failed to configure per-machine EXE package.
> [0F48:1064][2024-05-12T07:30:02]w350: Applied non-vital package: Redist2008_64, encountered error: 0x800700d8. Continuing...
> [0C14:057C][2024-05-12T07:30:11]e000: Error 0x800713ec: Process returned error: 0x13ec
> [0C14:057C][2024-05-12T07:30:11]e000: Error 0x800713ec: Failed to execute EXE package.
> [0F48:1064][2024-05-12T07:30:11]e000: Error 0x800713ec: Failed to configure per-machine EXE package.
> [0F48:1064][2024-05-12T07:30:11]w350: Applied non-vital package: Redist2010_64, encountered error: 0x800713ec. Continuing...
> [0C14:057C][2024-05-12T07:30:14]e000: Error 0x81f40001: Process returned error: 0x81f40001
> [0C14:057C][2024-05-12T07:30:14]e000: Error 0x81f40001: Failed to execute EXE package.
> [0F48:1064][2024-05-12T07:30:14]e000: Error 0x81f40001: Failed to configure per-machine EXE package.
> [0F48:1064][2024-05-12T07:30:14]w350: Applied non-vital package: Redist2012_64, encountered error: 0x81f40001. Continuing...
> [0C14:057C][2024-05-12T07:30:18]e000: Error 0x80070001: Process returned error: 0x1
> [0C14:057C][2024-05-12T07:30:18]e000: Error 0x80070001: Failed to execute EXE package.
> [0F48:1064][2024-05-12T07:30:18]e000: Error 0x80070001: Failed to configure per-machine EXE package.
> [0F48:1064][2024-05-12T07:30:18]w350: Applied non-vital package: Redist2013_64, encountered error: 0x80070001. Continuing...
> [0C14:057C][2024-05-12T07:30:21]e000: Error 0x80070001: Process returned error: 0x1
> [0C14:057C][2024-05-12T07:30:21]e000: Error 0x80070001: Failed to execute EXE package.
> [0F48:1064][2024-05-12T07:30:21]e000: Error 0x80070001: Failed to configure per-machine EXE package.
> [0F48:1064][2024-05-12T07:30:21]w350: Applied non-vital package: Redist2015_64, encountered error: 0x80070001. Continuing...
> [0C14:057C][2024-05-12T07:30:27]e000: Error 0x80070001: Process returned error: 0x1
> [0C14:057C][2024-05-12T07:30:27]e000: Error 0x80070001: Failed to execute EXE package.
> [0F48:1064][2024-05-12T07:30:27]e000: Error 0x80070001: Failed to configure per-machine EXE package.
> [0F48:1064][2024-05-12T07:30:27]w350: Applied non-vital package: Redist2019_64, encountered error: 0x80070001. Continuing...

(Automated response - build 880.)

File: manifests/r/RaptorTech/RaptorHardwareService/2.3.6.0/RaptorTech.RaptorHardwareService.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.6.1.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: RaptorTech.RaptorHardwareService
+PackageVersion: 2.3.6.0
+InstallerType: burn
+Installers:
```
Comment (stephengillie) [inline manifests/r/RaptorTech/RaptorHardwareService/2.3.6.0/RaptorTech.RaptorHardwareService.installer.yaml:7]:
```suggestion
Dependencies:
  WindowsFeatures:
    - netfx3 
Installers:
```

**Comment by stephengillie** [comment]
With the dependency installed, this package installs successfully, with exit code `-1978335226` whic often means "Install successful but system restart may be required."

Verify:
image

This package also installs these other packages:
- `3M(TM)) Full Page Reader Device Drivers` (thus)
- Acuant SDK
- Bonjour (Apple networking?)
- DYMO Connect
- DYMO Label v.8
- `VCRedist` 2008, 2010, 2012, 2015, and 2015-2019 - both `x86` and `x64` of each.
- Raptor Hardware Service Updater
- ScanShell 800DX
- ScanShell 800DX Driver

**Comment by chrisg32** [comment]
>
> This package also installs these other packages:
> 
>     * `3M(TM)) Full Page Reader Device Drivers` (thus)
> 
>     * Acuant SDK
> 
>     * Bonjour (Apple networking? AKA `Apple.Bonjour`)
> 
>     * DYMO Connect (`SanfordLP.DYMOConnect`)
> 
>     * DYMO Label v.8 (`SanfordLP.DYMOLabel`)
> 
>     * `VCRedist` 2008, 2010, 2012, and 2015-2019 - both `x86` and `x64` of each. (`Microsoft.VCRedist.`Year.Arch)
> 
>     * Raptor Hardware Service Updater
> 
>     * ScanShell 800DX
> 
>     * ScanShell 800DX Driver

That is expected. This software has dependencies on all those packages. Please let me know if I need to make any changes to allow for that.

I appreciate the help/patience. This is my first winget package.

**Comment by stephengillie** [comment]
> > This package also installs these other packages:
> > ```
> > * `3M(TM)) Full Page Reader Device Drivers` (thus)
> > 
> > * Acuant SDK
> > 
> > * Bonjour (Apple networking? AKA `Apple.Bonjour`)
> > 
> > * DYMO Connect (`SanfordLP.DYMOConnect`)
> > 
> > * DYMO Label v.8 (`SanfordLP.DYMOLabel`)
> > 
> > * `VCRedist` 2008, 2010, 2012, and 2015-2019 - both `x86` and `x64` of each. (`Microsoft.VCRedist.`Year.Arch)
> > 
> > * Raptor Hardware Service Updater
> > 
> > * ScanShell 800DX
> > 
> > * ScanShell 800DX Driver
> > ```
> 
> That is expected. This software has dependencies on all those packages. Please let me know if I need to make any changes to allow for that.
> 
> I appreciate the help/patience. This is my first winget package.

No worries here. To proceed, the .NET 3.5 Framework `WindowsFeature` [mentioned above](https://github.com/microsoft/winget-pkgs/pull/153243#discussion_r1599156276) will need to be added, since the program depends on it to run. This will automatically install the .NET 3.5 Framework before running this package's installer, so the framework is available.

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-05-19 16:38:20.803 [CLI ] ShellExecute installer failed: 1
> MSI (s) (AC:2C) [16:37:58:207]: APPCOMPAT: [DetectVersionLaunchCondition] Failed to initialize pRecErr.
> WixRemoveFoldersEx:  Error 0x80070057: Missing folder property: SUBINSTALLLOCATION for row: wrf726A0F9DBD80AD9854A373F3222ECCBF
> CustomAction WixRemoveFoldersEx returned actual error code 1603 but will be translated to success due to continue marking
> SchedSecureObjectsRollback:  Failed to store ACL rollback information with error 0x80070424 - continuing
> MSI (s) (AC:2C) [16:38:01:145]: Executing op: ComponentRegister(ComponentId={8C5E0981-4C64-5E8E-99D0-6F1F8B756800},KeyPath=C:\Program Files\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=0)
> MSI (s) (AC:2C) [16:38:02:145]: Executing op: FileCopy(SourceName=mdpzc2t8.dll|ErrorHandlerDll.dll,SourceCabKey=fil075DCCE23C502655966A6DE9112B3DF8,DestName=ErrorHandlerDll.dll,Attributes=512,FileSize=80384,PerTick=65536,,VerifyMedia=1,,,,,CheckCRC=0,Version=9.3.0.9,Language=2057,InstallMode=58982400,,,,,,,)
> MSI (s) (AC:2C) [16:38:02:145]: File: C:\Program Files\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll;	To be installed;	Won't patch;	No existing file
> MSI (s) (AC:2C) [16:38:07:317]: Executing op: ServiceInstall(Name=RaptorHardwareService,DisplayName=Raptor Hardware Service,ImagePath="C:\Program Files\Raptor Technologies LLC\RaptorHardwareService\Raptorware.ClientService.Server.exe",ServiceType=32,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=Service to facilitate scanning and printing for the Raptor System,,)
> MSI (s) (AC:08) [16:38:15:473]: APPCOMPAT: [DetectVersionLaunchCondition] Failed to initialize pRecErr.
> MSI (s) (AC:08) [16:38:17:004]: Executing op: ServiceInstall(Name=RaptorHardwareServiceUpdater,DisplayName=Raptor Hardware Service Updater,ImagePath="C:\Program Files\Raptor Technologies LLC\RaptorHardwareServiceUpdater\Raptor.Client.HardwareUpdaterService.exe",ServiceType=32,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=Service to update the Raptor System Hardware Service,,)
> [0830:11D8][2024-05-19T16:32:52]e000: Error 0x800700d8: Process returned error: 0xd8
> [0830:11D8][2024-05-19T16:32:52]e000: Error 0x800700d8: Failed to execute EXE package.
> [1CD8:0D94][2024-05-19T16:32:52]e000: Error 0x800700d8: Failed to configure per-machine EXE package.
> [1CD8:0D94][2024-05-19T16:32:52]w350: Applied non-vital package: Redist2008_64, encountered error: 0x800700d8. Continuing...
> [0830:11D8][2024-05-19T16:33:01]e000: Error 0x800713ec: Process returned error: 0x13ec
> [0830:11D8][2024-05-19T16:33:01]e000: Error 0x800713ec: Failed to execute EXE package.
> [1CD8:0D94][2024-05-19T16:33:01]e000: Error 0x800713ec: Failed to configure per-machine EXE package.
> [1CD8:0D94][2024-05-19T16:33:01]w350: Applied non-vital package: Redist2010_64, encountered error: 0x800713ec. Continuing...
> [0830:11D8][2024-05-19T16:33:04]e000: Error 0x81f40001: Process returned error: 0x81f40001
> [0830:11D8][2024-05-19T16:33:04]e000: Error 0x81f40001: Failed to execute EXE package.
> [1CD8:0D94][2024-05-19T16:33:04]e000: Error 0x81f40001: Failed to configure per-machine EXE package.
> [1CD8:0D94][2024-05-19T16:33:04]w350: Applied non-vital package: Redist2012_64, encountered error: 0x81f40001. Continuing...
> [0830:11D8][2024-05-19T16:33:07]e000: Error 0x80070001: Process returned error: 0x1
> [0830:11D8][2024-05-19T16:33:07]e000: Error 0x80070001: Failed to execute EXE package.
> [1CD8:0D94][2024-05-19T16:33:07]e000: Error 0x80070001: Failed to configure per-machine EXE package.
> [1CD8:0D94][2024-05-19T16:33:07]w350: Applied non-vital package: Redist2013_64, encountered error: 0x80070001. Continuing...
> [0830:11D8][2024-05-19T16:33:11]e000: Error 0x80070001: Process returned error: 0x1
> [0830:11D8][2024-05-19T16:33:11]e000: Error 0x80070001: Failed to execute EXE package.
> [1CD8:0D94][2024-05-19T16:33:11]e000: Error 0x80070001: Failed to configure per-machine EXE package.
> [1CD8:0D94][2024-05-19T16:33:11]w350: Applied non-vital package: Redist2015_64, encountered error: 0x80070001. Continuing...
> [0830:11D8][2024-05-19T16:33:17]e000: Error 0x80070001: Process returned error: 0x1
> [0830:11D8][2024-05-19T16:33:17]e000: Error 0x80070001: Failed to execute EXE package.
> [1CD8:0D94][2024-05-19T16:33:17]e000: Error 0x80070001: Failed to configure per-machine EXE package.
> [1CD8:0D94][2024-05-19T16:33:17]w350: Applied non-vital package: Redist2019_64, encountered error: 0x80070001. Continuing...

Automated error analysis suggests 0x80070001 may mean Windows cannot install required files.

Automated error analysis suggests 0x80070057 may mean The parameter is incorrect.

Automated error analysis suggests 1603 may mean Install successful but system restart may be required. Should this be add as an `InstallerSuccessCode` or similar?

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

(Automated response - build 881.)

File: manifests/r/RaptorTech/RaptorHardwareService/2.3.6.0/RaptorTech.RaptorHardwareService.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+PackageIdentifier: RaptorTech.RaptorHardwareService
+PackageVersion: 2.3.6.0
+InstallerType: burn
+Dependencies:
+  WindowsFeatures:
+    - netfx3 
+Installers:
+- InstallerUrl: https://cdn.raptortech.com/installers/stable/2_3_6_0/Raptor.HardwareServices_2_3_6_0.exe
+  Architecture: x86
```
Comment (stephengillie) [inline manifests/r/RaptorTech/RaptorHardwareService/2.3.6.0/RaptorTech.RaptorHardwareService.installer.yaml:12]:
```suggestion
  Architecture: x64
```

Our `x86` validations are performed on 32-bit OSes. From the errors, I'm guessing that either the package or its dependency are 64-bit only.

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-05-22 13:18:09.660 [CLI ] ShellExecute installer failed: 1
> MSI (s) (A8:7C) [13:17:45:854]: APPCOMPAT: [DetectVersionLaunchCondition] Failed to initialize pRecErr.
> WixRemoveFoldersEx:  Error 0x80070057: Missing folder property: SUBINSTALLLOCATION for row: wrf726A0F9DBD80AD9854A373F3222ECCBF
> CustomAction WixRemoveFoldersEx returned actual error code 1603 but will be translated to success due to continue marking
> SchedSecureObjectsRollback:  Failed to store ACL rollback information with error 0x80070424 - continuing
> MSI (s) (A8:7C) [13:17:49:026]: Executing op: ComponentRegister(ComponentId={8C5E0981-4C64-5E8E-99D0-6F1F8B756800},KeyPath=C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=0)
> MSI (s) (A8:7C) [13:17:49:026]: WIN64DUALFOLDERS: Substitution in 'C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll' folder had been blocked by the 1 mask argument (the folder pair's iSwapAttrib member = 0).
> MSI (s) (A8:7C) [13:17:50:120]: Executing op: FileCopy(SourceName=mdpzc2t8.dll|ErrorHandlerDll.dll,SourceCabKey=fil075DCCE23C502655966A6DE9112B3DF8,DestName=ErrorHandlerDll.dll,Attributes=512,FileSize=80384,PerTick=65536,,VerifyMedia=1,,,,,CheckCRC=0,Version=9.3.0.9,Language=2057,InstallMode=58982400,,,,,,,)
> MSI (s) (A8:7C) [13:17:50:120]: File: C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll;	To be installed;	Won't patch;	No existing file
> MSI (s) (A8:7C) [13:17:55:386]: Executing op: ServiceInstall(Name=RaptorHardwareService,DisplayName=Raptor Hardware Service,ImagePath="C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\Raptorware.ClientService.Server.exe",ServiceType=32,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=Service to facilitate scanning and printing for the Raptor System,,)
> MSI (s) (A8:74) [13:18:03:073]: APPCOMPAT: [DetectVersionLaunchCondition] Failed to initialize pRecErr.
> MSI (s) (A8:74) [13:18:05:542]: Executing op: ServiceInstall(Name=RaptorHardwareServiceUpdater,DisplayName=Raptor Hardware Service Updater,ImagePath="C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareServiceUpdater\Raptor.Client.HardwareUpdaterService.exe",ServiceType=32,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=Service to update the Raptor System Hardware Service,,)

Automated error analysis suggests 0x80070057 may mean The parameter is incorrect.

Automated error analysis suggests 0x80070424 may mean The specified service does not exist as an installed service.

Automated error analysis suggests 1603 may mean Install successful but system restart may be required. Should this be add as an `InstallerSuccessCode` or similar?

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

(Automated response - build 881.)

**Comment by stephengillie** [comment]
This package installs and launches correctly, despite returning a non-zero exit code.

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-05-30 06:00:18.618 [CLI ] ShellExecute installer failed: 1
> MSI (s) (D0:58) [05:59:55:121]: APPCOMPAT: [DetectVersionLaunchCondition] Failed to initialize pRecErr.
> WixRemoveFoldersEx:  Error 0x80070057: Missing folder property: SUBINSTALLLOCATION for row: wrf726A0F9DBD80AD9854A373F3222ECCBF
> CustomAction WixRemoveFoldersEx returned actual error code 1603 but will be translated to success due to continue marking
> SchedSecureObjectsRollback:  Failed to store ACL rollback information with error 0x80070424 - continuing
> MSI (s) (D0:58) [05:59:58:356]: Executing op: ComponentRegister(ComponentId={8C5E0981-4C64-5E8E-99D0-6F1F8B756800},KeyPath=C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll,State=3,,Disk=1,SharedDllRefCount=0,BinaryType=0)
> MSI (s) (D0:58) [05:59:58:356]: WIN64DUALFOLDERS: Substitution in 'C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll' folder had been blocked by the 1 mask argument (the folder pair's iSwapAttrib member = 0).
> MSI (s) (D0:58) [05:59:59:465]: Executing op: FileCopy(SourceName=mdpzc2t8.dll|ErrorHandlerDll.dll,SourceCabKey=fil075DCCE23C502655966A6DE9112B3DF8,DestName=ErrorHandlerDll.dll,Attributes=512,FileSize=80384,PerTick=65536,,VerifyMedia=1,,,,,CheckCRC=0,Version=9.3.0.9,Language=2057,InstallMode=58982400,,,,,,,)
> MSI (s) (D0:58) [05:59:59:465]: File: C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\ErrorHandlerDll.dll;	To be installed;	Won't patch;	No existing file
> MSI (s) (D0:58) [06:00:04:746]: Executing op: ServiceInstall(Name=RaptorHardwareService,DisplayName=Raptor Hardware Service,ImagePath="C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareService\Raptorware.ClientService.Server.exe",ServiceType=32,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=Service to facilitate scanning and printing for the Raptor System,,)
> MSI (s) (D0:BC) [06:00:12:793]: APPCOMPAT: [DetectVersionLaunchCondition] Failed to initialize pRecErr.
> MSI (s) (D0:BC) [06:00:14:762]: Executing op: ServiceInstall(Name=RaptorHardwareServiceUpdater,DisplayName=Raptor Hardware Service Updater,ImagePath="C:\Program Files (x86)\Raptor Technologies LLC\RaptorHardwareServiceUpdater\Raptor.Client.HardwareUpdaterService.exe",ServiceType=32,StartType=2,ErrorControl=1,,Dependencies=[~],,,Password=**********,Description=Service to update the Raptor System Hardware Service,,)

Automated error analysis suggests 0x80070057 may mean The parameter is incorrect.

Automated error analysis suggests 0x80070424 may mean The specified service does not exist as an installed service.

Automated error analysis suggests 1603 may mean Install successful but system restart may be required. Should this be add as an `InstallerSuccessCode` or similar?

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

(Automated response - build 882.)

**Pull request closed without merging**